lc-hax/Scripts/Core/HaxCamera.cs

```diff
@@ -0,0 +1,162 @@
+        if (this.HaxCameraContainer is null) return;
+        if (this.HaxCamAudioListener is not AudioListener haxListener) return;
+        if (this.GetCamera() is not Camera camera) return;
+
+        this.HaxCameraContainer.SetActive(active);
+
+        if (!this.HaxCameraContainer.activeSelf) {
+            playerlistener.enabled = true;
+            startOfRound.audioListener = playerlistener;
+            startOfRound.activeCamera.enabled = true;
+        }
+        else {
+            this.CopyFromCamera(startOfRound.activeCamera.transform, camera, startOfRound.activeCamera);
+            startOfRound.activeCamera.enabled = false;
+            playerlistener.enabled = false;
+            startOfRound.audioListener = haxListener;
+        }
+    }
+
+    void LateUpdate() {
+        if (this.HaxCameraContainer is null) return;
+        if (this.KeyboardMovement is null) return;
+        if (this.HaxCameraContainer.activeSelf) {
+            this.KeyboardMovement.IsPaused = Helper.LocalPlayer is { isTypingChat: true };
+        }
+    }
+
+    internal void DisableCamera() {
+        if (PhantomMod.Instance is PhantomMod phantom) {
+            phantom.DisablePhantom();
+        }
+
+        this.HaxCameraContainer?.SetActive(false);
+    }
+
+    internal Camera? GetCamera() {
+        if (this.CustomCamera != null) return this.CustomCamera;
+
+        this.HaxCameraContainer ??= new GameObject("lc-hax Camera Parent");
+        Camera newCamera = this.HaxCameraContainer.AddComponent<Camera>();
+
+        this.HaxCamAudioContainer ??= new GameObject("lc-hax Audio Listener");
+        this.HaxCamAudioListener = this.HaxCamAudioContainer.GetComponent<AudioListener>();
+        this.HaxCamAudioListener ??= this.HaxCamAudioContainer.AddComponent<AudioListener>();
+        this.HaxCamAudioListener.transform.SetParent(this.HaxCameraContainer.transform, false);
+        this.HaxCamAudioListener.transform.localScale = new Vector3(0.8196f, 0.8196f, 0.8196f);
+        // zero both local pos and rot
+        this.HaxCamAudioListener.transform.localPosition = Vector3.zero;
+        this.HaxCamAudioListener.transform.localRotation = Quaternion.identity;
+
+        // Set HaxCamAudioContainer as a child of HaxCamContainer
+        this.HaxCamAudioContainer.transform.SetParent(this.HaxCameraContainer.transform, false);
+
+        this.KeyboardMovement = newCamera.GetComponent<KeyboardMovement>();
+        this.KeyboardMovement ??= newCamera.gameObject.AddComponent<KeyboardMovement>();
+
+        this.MousePan = newCamera.GetComponent<MousePan>();
+        this.MousePan ??= newCamera.gameObject.AddComponent<MousePan>();
+
+        this.MousePan.enabled = true;
+        this.KeyboardMovement.enabled = true;
+        this.HaxCamAudioListener.enabled = true;
+        this.HaxCameraContainer.SetActive(false);
+        this.CustomCamera = newCamera;
+        DontDestroyOnLoad(this.HaxCameraContainer);
```

Comment:
Is this necessary? HaxCamera is already set to not destroy on load in Loader.cs right?

Reply:
the container where the camera is not , the module is , but not the container.

Reply:
Is it possible to make the container a child of the module?

Reply:
is best not, because the container is programmed to stalk the player.

Reply:
Because we the module doesn't stalk the player, we have to expose `HaxCameraContainer` which is so unnecessary.